### PR TITLE
linter: enable SA4018 and SA4004

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/40786

Problem Summary:

### What is changed and how it works?

SA4004 - The loop exits unconditionally after one iteration
SA4018 - Self-assignment of variables

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Defined2014 — APPROVED** [review]

File: server/http_status.go [hunk 1]
```diff
@@ -466,7 +466,7 @@ func (s *Server) startStatusServerAndRPCServer(serverMux *http.ServeMux) {
 	service.RegisterChannelzServiceToServer(grpcServer)
 	if s.cfg.Store == "tikv" {
 		keyspaceName := config.GetGlobalKeyspaceName()
-		for {
+		func() {
```
Comment (lance6716) [inline server/http_status.go:469]: maybe we use `goto` or ignore the lint

**Review by lance6716 — COMMENTED** [review]
rest lgtm

**Review by lance6716 — APPROVED** [review]

File: server/http_status.go (hunk 1 above)
Comment (hawkingrei) [reply server/http_status.go]: Done

**Pull request merged**